File: geo_obj.py
class GeoObject:

  # initialize object with all properties from CSV file
  def __init__(self, fid, area, perimeter, id, land, name, status, rb, type, date, layer, path, gen, kreis_kenn, z_kreis_ke, kreis_id, rs, two_kreis_ke,state, landkey_pa, rb_codes_r, longitude, latitude):
    self.fid = fid
    self.area = area
    self.perimeter = perimeter
    self.id = id
    self.land = land
    self.name = name
    self.status = status
    self.rb = rb
    self.type = type
    self.date = date
    self.layer = layer
    self.path = path
    self.gen = gen
    self.kreis_kenn = kreis_kenn
    self.z_kreis_ke = z_kreis_ke
    self.kreis_id = kreis_id
    self.rs = rs
    self.two_kreis_ke = two_kreis_ke
    self.state = state
    self.landkey_pa = landkey_pa
    self.rb_codes_r = rb_codes_r
    self.longitude = longitude
    self.latitude = latitude
    # these we will add later
    self.start_date = None
    self.end_date = None
    self.date_ranges = None

  # instance method to return values of all object properties above
  def to_csv(self):
    return [*self.__dict__.values()]

File: geo_obj.rb
class GeoObject
  attr_reader :f_id, :area, :perimeter, :id, :land, :name, :status, :rb, :type, :date, :layer, :path, :gen, :kreis_kenn, :z_kreis_ke, :kreis_id, :rs, :two_kreis_ke,:state, :landkey_pa, :rb_codes_r, :longitude, :latitude

  attr_accessor :start_date, :end_date, :date_ranges

  def initialize(args = {})
    @f_id = args[:f_id]
    @area = args[:area]
    @perimeter = args[:perimeter]
    @id = args[:id]
    @land = args[:land]
    @name = args[:name]
    @status = args[:status]
    @rb = args[:rb]
    @type = args[:type]
    @date = args[:date]
    @layer = args[:layer]
    @path = args[:path]
    @gen = args[:gen]
    @kreis_kenn = args[:kreis_kenn]
    @z_kreis_ke = args[:z_kreis_ke]
    @kreis_id = args[:kreis_id]
    @rs = args[:rs]
    @two_kreis_ke = args[:two_kreis_ke]
    @state = args[:state]
    @landkey_pa = args[:landkey_pa]
    @rb_codes_r = args[:rb_codes_r]
    @longitude = args[:longitude]
    @latitude = args[:latitude]
    @start_date = nil
    @end_date = nil
    @date_ranges = nil
  end

  def to_csv
    csv_array = []
    self.instance_variables.each do |attribute|
      csv_array.push(self.instance_variable_get(attribute))
    end
    csv_array
  end

end